extensions/control-plane/api/management-api/asset-api/src/main/java/org/eclipse/edc/connector/api/management/asset/AssetNewApiController.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    @POST
+    @Override
+    public IdResponseDto createAsset(@Valid AssetEntryNewDto assetEntryDto) {
+        var expandedJson = JsonLdUtil.expand(assetEntryDto.getAsset());
+        if (expandedJson.size() == 0) throw new InvalidRequestException("Invalid asset!");
+
+        var expandedJsonAsset = expandedJson.getJsonObject(0);
+        var assetResult = transformerRegistry.transform(expandedJsonAsset, Asset.class);
+        var dataAddressResult = transformerRegistry.transform(assetEntryDto.getDataAddress(), DataAddress.class);
+
+        if (assetResult.failed() || dataAddressResult.failed()) {
+            var errorMessages = Stream.concat(assetResult.getFailureMessages().stream(), dataAddressResult.getFailureMessages().stream());
+            throw new InvalidRequestException(errorMessages.collect(toList()));
+        }
+
+        var dataAddress = dataAddressResult.getContent();
+        var asset = assetResult.getContent();
+
+        var resultContent = service.create(asset, dataAddress).orElseThrow(exceptionMapper(Asset.class, asset.getId()));
+
+        return IdResponseDto.Builder.newInstance()
+                .id(resultContent.getId())
+                .createdAt(resultContent.getCreatedAt())
+                .build();
+    }
+
+    @POST
+    @Path("/request")
+    @Override
+    public List<AssetResponseNewDto> requestAssets(@Valid QuerySpecDto querySpecDto) {
+        return queryAssets(ofNullable(querySpecDto).orElse(QuerySpecDto.Builder.newInstance().build()));
+    }
+
+    @GET
+    @Path("{id}")
+    @Override
+    public AssetResponseNewDto getAsset(@PathParam("id") String id) {
+        var obj = of(id)
+                .map(it -> service.findById(id))
+                .map(it -> transformerRegistry.transform(it, JsonObject.class))
+                .filter(Result::succeeded)
+                .map(Result::getContent)
+                .orElseThrow(() -> new ObjectNotFoundException(Asset.class, id));
+        return Optional.ofNullable(expand(obj))
+                .orElseThrow(() -> new EdcException("Error expanding asset"));
+    }
+
+    @DELETE
+    @Path("{id}")
+    @Override
+    public void removeAsset(@PathParam("id") String id) {
+        service.delete(id).orElseThrow(exceptionMapper(Asset.class, id));
+    }
+
+    @PUT
+    @Path("{assetId}")
+    @Override
+    public void updateAsset(@PathParam("assetId") String assetId, @Valid AssetUpdateRequestDto asset) {
+        var wrapper = AssetUpdateRequestWrapperDto.Builder.newInstance().updateRequest(asset).assetId(assetId).build();
+        var assetResult = transformerRegistry.transform(wrapper, Asset.class);
+        if (assetResult.failed()) {
+            throw new InvalidRequestException(assetResult.getFailureMessages());
+        }
+        service.update(assetResult.getContent())
+                .orElseThrow(exceptionMapper(Asset.class, assetId));
+    }
+
+    @PUT
+    @Path("{assetId}/dataaddress")
```

Comment:
here's `dataaddress` and at 151 it is `address`. Personally I prefer the simplest `address` one

Reply:
i agree on making it consistent, however, i'd put `dataaddress`, simply because that is what the object is called. In my mind consistency beats simplicity here :)